domain/model/state/controllerstate.go

```diff
@@ -1802,3 +1803,50 @@ AND    ot.type = $dbPermission.object_type
 	}
 	return nil
 }
+
+// GetAllActivatedModelsUUIDQuery returns a SQL statement that will get all the activated models uuids in the controller.
+func (st *State) GetAllActivatedModelsUUIDQuery() string {
+	return "SELECT uuid from model WHERE activated = true"
+}
+
+// GetActivatedModelUUIDs returns the subset of model uuids from the supplied list that are activated.
+// If no models associated with the uuids are activated then an empty slice is returned.
+func (st *State) GetActivatedModelUUIDs(ctx context.Context, uuids []coremodel.UUID) ([]coremodel.UUID, error) {
+	if len(uuids) == 0 {
+		return nil, nil
+	}
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	type modelUUID struct {
+		UUID coremodel.UUID `db:"uuid"`
+	}
+
+	stmt, err := st.Prepare(`
+SELECT uuid AS &modelUUID.uuid
+FROM   v_model
+WHERE uuid IN ($S[:])
+`, sqlair.S{}, modelUUID{})
+
+	if err != nil {
+		return nil, errors.Errorf("preparing select model activated status statement: %w", err)
```

Comment:
We have broad agreement that returning `errors.Capture` is enough for these types of errors that we don't expect to happen in production code.

Reply:
Sure thanks tom!